internal/testrunner/runners/system/runner.go

```diff
@@ -683,14 +725,57 @@ func (r *runner) prepareScenario(config *testConfig, ctxt servicedeployer.Servic
 	if err != nil {
 		return nil, fmt.Errorf("failed to find the selected policy_template: %w", err)
 	}
+	// Setup agent
+	logger.Debug("setting up agent...")
+	agentOptions := r.createAgentOptions(scenario)
+	agentInfo, err := r.createAgentInfo(scenario)
+	if err != nil {
+		return nil, err
+	}
+
+	// FIXME: this should be stored in service state or not used in
+	// --no-provision and --tear-down stages
+	enrollingTime := time.Now()
```

Comment:
To be tested if it can be used Elastic Agent tags to be able to select the right agent without depending on the enrolling time or the order in the response. Currently, it selects the first agent (item) in the response.